src/snowflake/snowpark/session.py

```diff
@@ -1719,7 +1719,26 @@ def _get_dependency_packages(
                         package_client_version = pkg_resources.get_distribution(
                             package_name
                         ).version
-                        if package_client_version not in valid_packages[package_name]:
+
```

Comment:
maybe add a comment explaining why minor version is not considered

Reply:
updated